src/EnergyPlus/DataHVACGlobals.hh

```diff
@@ -100,10 +100,17 @@ namespace DataHVACGlobals {
     int constexpr CycleOn(2);
     int constexpr CycleOnZoneFansOnly(3);
     // The following parameters describe the setpoint types in TempControlType(ActualZoneNum)
-    int constexpr SingleHeatingSetPoint(1);
-    int constexpr SingleCoolingSetPoint(2);
-    int constexpr SingleHeatCoolSetPoint(3);
-    int constexpr DualSetPointWithDeadBand(4);
+    enum class SetPointType
```

Comment:
It might be more clear to call this `enum class ThermostatType` or `ThermostatSetPointType`?  `SetpointType` could be confused with the control variables in setpoint managers, and with the setpoint manager types, and with plantloop setpoint types.

Reply:
@mjwitte what about this madness? https://github.com/NREL/EnergyPlus/blob/c4bc5e27e0c333200eccfa55a1cf0fefba947b68/src/EnergyPlus/ZoneTempPredictorCorrector.hh#L102-L123

This `SetPointType` class is being used with variables that are also compared against the `ComfortControl` enum class.

Reply:
Hmm, yeah, this object:
```
ZoneControl:Thermostat:ThermalComfort,
  A5 , \field Thermal Comfort Control Type Schedule Name
       \note The Thermal Comfort Control Type Schedule contains values that are appropriate control types.
       \note Thermal Comfort Control types are integers: 0 - Uncontrolled (floating),
       \note 1 = ThermostatSetpoint:ThermalComfort:Fanger:SingleHeating
       \note 2 = ThermostatSetpoint:ThermalComfort:Fanger:SingleCooling
       \note 3 = ThermostatSetpoint:ThermalComfort:Fanger:SingleHeatingOrCooling
       \note 4 = ThermostatSetpoint:ThermalComfort:Fanger:DualSetpoint
```
I don't think the Pierce and KSU options were ever implemented, based on a quick search I think those can be deleted, along with the pierce and KSU variables in `DataZoneControls`.

But here, I don't think Uncontrolled thru DualSetPoint belong in `ComfortControl` I haven't read the code carefully, but those should only be valid for the basic thermostat, so they should be using your new enum class. These shouldn't have been blended into one dual-use enum here. At least I don't think so based on 10 minutes of code skimming.

Reply:
Just a note on this "madness". At some point in the not too distant future, we will be moving to a scenario where we are automatically generating the `enum class` definitions from the IDD/epJSON-schema file at build time the way OpenStudio does now (as an aside we will also be automatically generating some of the object definition code and the code that is now in `getInput`, it's going to be bananas). But under this scenario, these types of shenanigans (i.e., defining multiple names for the same value within an enumeration or really any type of `enum class` shenanigan including things such as starting valid enumerations at 1 will not be allowed.